### DISCUSSION TITLE
How to enable automplete for Database fields?

### DISCUSSION THREAD

**Discussion by Shadenight123** [vscode-abl/vscode-abl#133] [Q&A] [answered]
I apologize if this is the most dumb question ever asked, but I've looked pretty much everywhere (except perhaps the big red bolded line which states how to enable this) and I cannot seem to find how to enable autocomplete for what concerns the database fields and so forth.
To be extra clear:
if I have a table known as "articles" with the fields "name" "date" and "price", I'd expect there to be, when I start typing 'articles' the following:
articles.name
articles.date
articles.price 
and the possibility to pick one of these (so, autocomplete in a nutshell).
However both the autocomplete feature and the 'Read dictionary' feature (which apparently is supposed to be there) aren't there in the command palette.
What am I missing? I used the suggested script to grab the configuration files from where I'm usually working, and it *is* working; when I try some code on VS, it does correctly run, identifies the tables and outputs correctly.
The autocomplete just doesn't work, so I'm stumped on the reasons why.
Thank you for your patience and your time!

**Comment by PeterJudgeZA** [comment, marked as answer]
The trick is to have a DF in the database connections - you need the dumpFile property set to the db's DF.

```
{
      "name": "sp2k", // Logical name of database, or physical name if -ld is not used
      "connect": "-db db/sp2k -RO", // ABL connection string
      "dumpFile": "dump/sp2k.df",   // Database schema, DF format
      "aliases": [ "foo", "bar"]    // List of aliases in ABL session
    }
```

**Reply by Shadenight123** [reply]
Thank you so much for this!
Basically, what I failed to understand was that "dumpfile" had to be a literal 'dump' folder with the DBNAME.df inside of it within the root of VSCODE folder.
So, for everyone else who might be equally stumped by such an obvious thing, I'll say this:
1) Dump the database scheme (dbname.df) of your tables. 
2) move it to the 'Dump' folder (if one isn't there, create it yourself) of the vscode work area.
3) ensure "dumpFile": has "dump/dbname.df"
4) restart vscode.
5) Enjoy autocomplete.

Thank you so much once more!

**Reply by gquerret** [reply]
Attribute name will be improved
